src/build-time-render/BuildTimeRender.ts

```diff
@@ -148,68 +152,79 @@ export default class BuildTimeRender {
+						writtenAssets.push(additionalChunk);
+
+						html = html.replace(
+							'</body>',
+							`<link rel="preload" href="${additionalChunk}" as="script"></body>`
+						);
+					});
 
-		let css = this._entries.reduce((css, entry) => {
-			const cssFile = this._manifest[entry.replace('.js', '.css')];
-			if (cssFile) {
-				html = html.replace(`<link href="${cssFile}" rel="stylesheet">`, '');
-				css = `${css}<link rel="stylesheet" href="${cssFile}" />`;
-
-				writtenAssets.push(cssFile);
+				Object.keys(this._manifest)
+					.filter((name) => name.endsWith('.js') || name.endsWith('.css'))
+					.filter((name) => !name.startsWith('runtime/'))
+					.filter((name) => !writtenAssets.some((asset) => this._manifest[name] === asset))
+					.forEach((preload) => {
+						html = html.replace(
+							'</body>',
+							`<link rel="prefetch" href="${this._manifest[preload].replace(
+								this._baseUrl.slice(1),
+								''
+							)}" /></body>`
+						);
+					});
 			}
-			return css;
-		}, '');
-
-		css = additionalCss.reduce((prev, url) => {
-			url = url.replace(this._baseUrl.slice(1), '');
-			writtenAssets.push(url);
-
-			return `${prev}<link rel="preload" href="${url}" as="style">`;
-		}, css);
-
-		styles = await this._processCss(styles);
-		html = html.replace(`</head>`, `<style>${styles}</style></head>`);
-		if (this._static || staticPath) {
-			html = html.replace(this._createScripts(), '');
 		} else {
-			html = html.replace(this._createScripts(), `${script}${css}${this._createScripts(false)}`);
-			blockScripts.forEach((blockScript, i) => {
-				writtenAssets.push(blockScript);
-
-				html = html.replace('</body>', `<script type="text/javascript" src="${blockScript}"></script></body>`);
-			});
-
-			const mainScript = this._manifest['main.js'];
-
-			additionalScripts
-				.sort((script1, script2) => {
-					return script1 === mainScript && !(script2 === mainScript) ? 1 : -1;
-				})
-				.forEach((additionalChunk: string) => {
-					additionalChunk = additionalChunk.replace(this._baseUrl.slice(1), '');
-					writtenAssets.push(additionalChunk);
-
-					html = html.replace('</body>', `<link rel="preload" href="${additionalChunk}" as="script"></body>`);
-				});
-
-			Object.keys(this._manifest)
-				.filter((name) => name.endsWith('.js') || name.endsWith('.css'))
-				.filter((name) => !name.startsWith('runtime/'))
-				.filter((name) => !writtenAssets.some((asset) => this._manifest[name] === asset))
-				.forEach((preload) => {
-					html = html.replace(
-						'</body>',
-						`<link rel="prefetch" href="${this._manifest[preload].replace(
-							this._baseUrl.slice(1),
-							''
-						)}" /></body>`
-					);
-				});
+			if (!this._sync) {
+				html = html.replace(
+					'</body>',
+					`<script type="text/javascript" src="${this._manifest['runtime/blocks.js']}"></script></body>`
+				);
+			}
 		}
+		blockScripts.forEach((blockScript, i) => {
```

Comment:
We always want to add each of the blocks chunks for normal and blocksOnly mode. On sync `blocksScripts` will be empty do not need any conditional logic based on that.